statick_tool/plugins/tool/spotbugs_tool_plugin.py

```diff
@@ -0,0 +1,119 @@
+import os
+import subprocess
+import xml.etree.ElementTree as etree
+
+from statick_tool.issue import Issue
+from statick_tool.tool_plugin import ToolPlugin
+
+
+class SpotbugsToolPlugin(ToolPlugin):
+    """Apply spotbugs tool and gather results."""
+
+    def get_name(self):
+        """Get name of tool."""
+        return "spotbugs"
+
+    def get_tool_dependencies(self):
+        """Get a list of tools that must run before this one."""
+        return ["make"]
+
+    def scan(self, package, level):
+        """Run tool and gather output."""
+        # Sanity check - make sure mvn exists
+        if not self.command_exists('mvn'):
+            print("Couldn't find 'mvn' command, can't run Spotbugs Maven integration")
+            # Return [] instead of None because this isn't an error per se,
+            # and we don't want to fail because a tool is missing
+            return []
+
+        flags = ["-Dspotbugs.effort=Max", "-Dspotbugs.threshold=Low",
+                 "-Dspotbugs.xmlOutput=true"]
+        flags += self.get_user_flags(level)
+
+        include_file = self.plugin_context.config.get_tool_config(self.get_name(),
+                                                                  level, "include")
+        exclude_file = self.plugin_context.config.get_tool_config(self.get_name(),
+                                                                  level, "exclude")
+        if include_file is not None:
+            flags += ["-Dspotbugs.includeFilterFile={}".format(self.plugin_context
+                                                               .resources.get_file(include_file))]
+
+        if exclude_file is not None:
+            flags += ["-Dspotbugs.excludeFilterFile={}".format(self.plugin_context
+                                                               .resources.get_file(exclude_file))]
+
+        issues = []
+        with open(self.get_name() + ".log", "w") as f:
+            for pom in package['top_poms']:
+                try:
+                    # The spotbugs:spotbugs-maven-plugin split is auto-concatenated
+                    output = subprocess.check_output(["mvn", "com.github.spotbugs:"
+                                                      "spotbugs-maven-plugin:spotbugs"] +
+                                                     flags,
+                                                     cwd=os.path.dirname(pom),
+                                                     stderr=subprocess.STDOUT,
+                                                     universal_newlines=True)
+                except subprocess.CalledProcessError as ex:
+                    output = ex.output
+                    f.write(output)
+                    print("spotbugs failed! Returncode = {}".
+                          format(str(ex.returncode)))
+                    print("{}".format(ex.output))
+                    return None
+
+                except OSError as ex:
+                    print("Couldn't find maven! (%s)" % (ex))
```

Comment:
I don't know that we are entirely consistent, but changing from `%` to `.format` seems like it would be closer to the rest of our print statements.

Reply:
Will do.